src/NuGet.Core/NuGet.Configuration/Settings/Items/SourceItem.cs

```diff
@@ -23,13 +23,41 @@ public string ProtocolVersion
             set => AddOrUpdateAttribute(ConfigurationConstants.ProtocolVersionAttribute, value);
         }
 
-        public SourceItem(string key, string value, string protocolVersion = "")
+        public string AllowInsecureConnections
+        {
+            get
+            {
+                if (Attributes.TryGetValue(ConfigurationConstants.AllowInsecureConnections, out var attribute))
+                {
+                    return Settings.ApplyEnvironmentTransform(attribute);
+                }
+
+                return null;
+            }
+            set => AddOrUpdateAttribute(ConfigurationConstants.AllowInsecureConnections, value);
+        }
+
+        public SourceItem(string key, string value)
+            : this(key, value, protocolVersion: "", allowInsecureConnections: "")
+        {
+        }
+
+        public SourceItem(string key, string value, string protocolVersion)
+            : this(key, value, protocolVersion, allowInsecureConnections: "")
+        {
+        }
+
+        public SourceItem(string key, string value, string protocolVersion = "", string allowInsecureConnections = "")
```

Comment:
The optional parameters no longer provide value.  If someone has source code calling `new SourceItem(key, value)`, it's going to use the 2 string overload, not this one with the default parameters. Same for the 3 string `new Source(key, value, protocolVersion)` overload, it's not going to use this 4 string overload. Therefore, the only way to call this overload is to explicitly pass in all 4 parameters, at which point none of them are using the default values.

Well, that's not entirely true. It's possible to call `new SourceItem(key, value, allowInsecureConnections: "true")`, but since this is a new API, there's obviously not going to be any existing code doing this.

It also avoid potential future issues with public APIs and default parameter analyzer violations.

Reply:
Thanks! Fixed. Removed the public API with optional arguments.